pkg/image/skopeo.go

```diff
@@ -95,6 +95,61 @@ func CopyRegistryImage(url, dest, accessKey, secKey string) error {
 	return err
 }
 
+// CopyDirFromRegistryImage download image from registry with skopeo and extracts a specific directory from it
+func CopyDirFromRegistryImage(url, dest, VMimagePath, accessKey, secKey string) error {
+	skopeoDest := "dir:" + dest + dataTmpDir
+	err := SkopeoInterface.CopyImage(url, skopeoDest, accessKey, secKey)
+	if err != nil {
+		os.RemoveAll(dest + dataTmpDir)
+		return errors.Wrap(err, "Failed to download from registry")
+	}
+	err = extractVMDiskImageDir(dest, VMimagePath)
+	if err != nil {
+		return errors.Wrap(err, "Failed to extract image layers")
+	}
+
+	// Clean temp folder
+	os.RemoveAll(dest + dataTmpDir)
+
+	return err
+}
+
+//extractVMDiskImageDir - from each layer extracts only disk/ directory
+//Since this is the oly one that is in intereset right now
```

Comment:
* typos: 'oly' -> 'only', 'intereset' - 'interest'
* Maybe worth keeping the signature more generalised. I.e. 'extractImageDir = func(dest, imagePath string)'.

Reply:
agreed